### PR TITLE
Adds a number of examples for builtin functions

### PR DESCRIPTION
**Pull request by punkle**
Adds a number of examples and tests for builtin functions

This pull requests also proposes:
- changing the pattern by which the examples are imported

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by punkle** [comment]
contributes to https://github.com/backstage/backstage/issues/18832

**Comment by freben** [comment]
Those are odd CI errors. You may want to keep things rebased rather than merging master - might be a matter of taste of course.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.examples.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          name: 'Fetch catalog entity',
+          input: {
+            entityRef: 'component:default/name',
+          },
+        },
+      ],
+    }),
+  },
+  {
+    description: 'Fetch multiple entities by referencse',
```
Comment (steff-petro) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.examples.ts:37]: Reference is misspelled here, just small thing I noticed

Comment (punkle) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.examples.ts]: thanks, should be fixed now.

**Comment by punkle** [comment]
The CI errors were due to changes to imports. It broke the mocking and I didnt fix it right. It should be ok now.. :D

**Review by freben — APPROVED** [review]
NICE!

**Pull request merged**